arduino/arduino.ino

```
@@ -1,25 +1,63 @@
+#define DROPS_PER_INCH 96
+#define INCHES_PER_REV 1.5748 // 40 mm per revolution
+#define STEPS_PER_REV 3200    // 200 full steps * 16 microsteps
+#define STEPS_PER_DROP 21     // (1/96)*(1/1.5748)*3200 = 21.17
+#define NOZZLE_COUNT 1
+
+#define MOTOR_X_DIR_PIN 2
+#define MOTOR_X_STEP_PIN 3
+#define MOTOR_X_LIMIT_PIN 6
+#define MOTOR_Y_DIR_PIN 3
+#define MOTOR_Y_STEP_PIN 4
+#define MOTOR_Y_LIMIT_PIN 7
+
 enum CommandMode {
   INSTRUCTION,
   MASK1,
   MASK2,
 };
 
+struct StepperMotor {
+  int dirPin;
+  int stepPin;
+  int limitPin;
+};
+
 CommandMode cmdMode = INSTRUCTION;
+StepperMotor motorX = {MOTOR_X_DIR_PIN, MOTOR_X_STEP_PIN, MOTOR_X_LIMIT_PIN};
+StepperMotor motorY = {MOTOR_Y_DIR_PIN, MOTOR_Y_STEP_PIN, MOTOR_Y_LIMIT_PIN};
+
+int stepsTaken = 0;
 
-// move stepper x 1 step (dot width) to the right
+// move stepper 1 step (dot width) to the right
 // keep track of steps moved so we can return to the begin of line
-void step() {
-  // TODO
+void step(const StepperMotor& motor = motorX, int dir = 0) {
```

avoid default arguments, it obscures things and can cause confusion.